src/aind_metadata_upgrader/procedures_upgrade.py

```diff
@@ -0,0 +1,377 @@
+
+        new_materials = []
+        for injection_material in old_injection_materials: #this wont work like i want, we changed the naming convention
+            new_material = None
+
+            if injection_material.get("titer") is not None:
+
+                injection_material = pop_unused_fields(injection_material, ViralMaterial)
+                
+                new_material = ViralMaterial.model_construct(injection_material)
+            elif injection_material.get("concentration") is not None:
+
+                injection_material = pop_unused_fields(injection_material, NonViralMaterial)
+
+                new_material = NonViralMaterial.model_construct(injection_material)
+            else:
+                logging.error(f"Injection material with no titer or concentration {injection_material} passed in")
+
+            if new_material:
+                new_materials.append(new_material)
+        
+        return new_materials
+
+
+class SubjectProcedureModelsUpgrade:
+    """Handle upgrades for SubjectProcedure models."""
+
+
+    def upgrade_craniotomy(old_subj_procedure: dict):
+        """Map legacy Craniotomy model to current version"""
+
+        if not check_field(old_subj_procedure, "protocol_id"):
+            old_subj_procedure["protocol_id"] = "unknown"
+
+        try:
+            return Craniotomy.model_validate(old_subj_procedure)
+        except ValidationError as e:
+            logging.error(f"Error validating IntraperitonealInjection: {e}")
+            return Craniotomy.model_construct(**old_subj_procedure)
+    
+    def upgrade_fiber_implant(old_subj_procedure: dict):
+        """Map legacy FiberImplant model to current version"""
+        print("upgrading fiber implant")
+        logging.info("FIBER IMPLANT: ", old_subj_procedure.keys())
+
+        probes = []
+        # for probe in old_subj_procedure.get("probes", []):
+        #     logging.info("PROBE: ", probe)
+        #     for field in probe.keys():
+        #         logging.info(f'FIELD: {field}, Value: {probe[field]}')
+        #     logging.info(f"PROBE: {probe}")
+        if check_field(old_subj_procedure, "probes"):
+            logging.info("FOUND PROBES")
+            if isinstance(old_subj_procedure["probes"], dict):
+                logging.info("PROBES IS DICT")
+                for field in old_subj_procedure["probes"].keys():
+                    logging.info(f'FIELD: {field}, Value: {old_subj_procedure["probes"][field]}')
+                probes.append(OphysProbe.model_construct(old_subj_procedure["probes"]))
+            elif isinstance(old_subj_procedure["probes"], list):
+                logging.info("PROBES IS LIST")
+                for probe in old_subj_procedure["probes"]:
+                    for field in probe.keys():
+                        logging.info(f'FIELD: {field}, Value: {probe[field]}', )
+                    probe['core_diameter_unit'] = "um"
+                    drop_unused_fields(probe, "ophys_probe")
```

Comment:
Are you sure you want to be modifying the input this way? If you are going to do this, you should do it in a copy of the dictionary.